tensorboard/plugins/profile/overview_page/overview-page.html

```diff
@@ -0,0 +1,353 @@
+       this.set('_device_idle_time_percent', generalAnalysisJson.p['device_idle_time_percent']);
+       this.set('_mxu_utilization_percent', generalAnalysisJson.p['mxu_utilization_percent']);
+       this.set('_steptime_ms_average', inputAnalysisJson.p['steptime_ms_average']);
+       this.set('_steptime_ms_stddev', inputAnalysisJson.p['steptime_ms_standard_deviation']);
+       this.set('_error_message', runEnvironmentJson.p['error_message']);
+       this.set('_host_count', runEnvironmentJson.p['host_count']);
+       this.set('_tpu_type', runEnvironmentJson.p['tpu_type']);
+       this.set('_tpu_core_count', runEnvironmentJson.p['tpu_core_count']);
+       this.set('_batch_size', runEnvironmentJson.p['batch_size']);
+       this.set('_change_list', runEnvironmentJson.p['change_list']);
+       this.set('_build_time', runEnvironmentJson.p['build_time']);
+       this.set('_build_target', runEnvironmentJson.p['build_target']);
+       this.set('_statement', recommendationJson.p['statement']);
+       this.set('_show_error_message', this._error_message!='');
+       this.updateStyles();
+
+       this._showDeviceStepChart(inputAnalysisJson);
+       this._showTopOpsTable(generalAnalysisJson);
+       this._showRecommendation(recommendationJson);
+     },
+
+     /* Plots a graph of step time (in ms) against step number */
+     _showDeviceStepChart : function(deviceJson) {
+       var computeTimeSeries = [];
+       var stepTimeSeries = [];
+       var maxStepTime = 0;
+       var step = -1;
+       deviceJson.rows.forEach(function(row, index) {
+         step = step > 0 ? ++step : Number(row.c[0].v);
+         computeTimeSeries.push({"scalar": row.c[1].v,
+				 "step": step,
+				 "tpu_step": Number(row.c[0].v),
+				 "low_watermark": 0})
+         stepTimeSeries.push({"scalar": row.c[1].v + row.c[2].v,
+                              "step": step,
+                              "low_watermark": row.c[1].v})
+         maxStepTime = Math.max(maxStepTime, row.c[1].v + row.c[2].v)
+       })
+       var chart = this.$.device_step_chart;
+       if (chart) {
+         chart.setVisibleSeries(['compute-time', 'step-time = input-time + compute-time']);
+	 chart.setSeriesData('compute-time', computeTimeSeries);
+         chart.setSeriesData('step-time = input-time + compute-time', stepTimeSeries);
+         chart.defaultYRange = [0, maxStepTime * 1.1]
+         chart.smoothingEnabled = false;
+         chart.tooltipColumns = [
+           { title: 'Name', evaluate: (d) => d.dataset.metadata().name },
+           { title: 'Time(ms)', evaluate: (d) => (d.datum.scalar).toFixed(2) },
+           { title: 'Step', evaluate: (d) => d.datum.step }];
+         chart.fillArea = {
+           higherAccessor: (d) => d.scalar,
+           lowerAccessor: (d) => d.low_watermark,
+         };
+         chart.xAxisFormatter = d3.format('d');
+       }
+     },
+
+     /* Draws a table of TensorFlow Op statistics */
+     _showTopOpsTable: function(opsTableJson) {
```

Comment:
Thanks for figuring this out. Unfortunately, polymer does not support dom-repeat patterns within table elements.

Reply:
No problem.